src/shell/shell_cmds.cpp

```diff
@@ -2242,3 +2243,271 @@ void DOS_Shell::CMD_VOL(char* args)
 	const auto trimmed_label = trim(label);
 	WriteOut(MSG_Get("SHELL_CMD_VOL_OUTPUT"), drive, trimmed_label, high, low);
 }
+
+void DOS_Shell::CMD_MOVE(char* args)
+{
+	HELP("MOVE");
+
+	// TODO: Add support for these flags along with the COPYCMD environment
+	// variable (overwrite prompts) Also currently ignored by CMD_COPY
+	(void)ScanCMDBool(args, "Y");
+	(void)ScanCMDBool(args, "-Y");
+
+	char* rem = ScanCMDRemain(args);
+	if (rem) {
+		WriteOut(MSG_Get("SHELL_ILLEGAL_SWITCH"), rem);
+		return;
+	}
+
+	size_t num_sources = 0;
+	char* raw_sources[32];
+	bool parsing          = true;
+	bool quote            = false;
+	bool stop_next_letter = false;
+	bool comma            = false;
+	char* word            = args;
+
+	// Parse sources (seperated by commas)
+	while (parsing) {
+		if (num_sources >= ARRAY_LEN(raw_sources)) {
+			WriteOut(MSG_Get("SHELL_TOO_MANY_PARAMETERS"));
+			return;
+		}
+		switch (*args) {
+		case '"':
+			quote = !quote;
+			memmove(args, args + 1, strlen(args));
+			break;
+		case ' ':
+			if (quote) {
+				args++;
+			} else {
+				if (word != args) {
+					*args                      = 0;
+					raw_sources[num_sources++] = word;
+					stop_next_letter           = true;
+					comma                      = false;
+				}
+				args++;
+				word = args;
+			}
+			break;
+		case ',':
+			stop_next_letter = true;
+			comma            = true;
+			if (word != args) {
+				*args                      = 0;
+				raw_sources[num_sources++] = word;
+			}
+			args++;
+			word = args;
+			break;
+		case 0: parsing = false; break;
+		default:
+			if (stop_next_letter) {
+				parsing          = comma;
+				comma            = false;
+				stop_next_letter = false;
+			} else {
+				args++;
+			}
+		}
+	}
+
+	if (num_sources < 1) {
+		WriteOut(MSG_Get("SHELL_MISSING_PARAMETER"));
+		return;
+	}
+
+	parsing               = true;
+	char* raw_destination = args;
+
+	// Parse detination
+	while (parsing) {
+		switch (*args) {
+		case '"':
+			quote = !quote;
+			memmove(args, args + 1, strlen(args));
+			break;
+		case ' ':
+			if (quote) {
+				args++;
+			} else {
+				*args++ = 0;
+				parsing = false;
+			}
+			break;
+		case '\0': parsing = false; break;
+		default: args++;
+		}
+	}
+
+	if (!*raw_destination) {
+		WriteOut(MSG_Get("SHELL_MISSING_PARAMETER"));
+		return;
+	}
+
+	// Check for extra arguments
+	while (*args) {
```

Comment:
C-style argument array is prone to command line parser bugs. If you want to parse remaining arguments manually, I suggest to use vector of strings:

```c
std::vector<std::string> params;
cmd->FillVector(params);
...
```

Reply:
Also please use the highest level functions thst operate on std::strings. This is not performance intensive code, we should absolutely optimise for simplicity, succinctness, and ease of understanding.

Look into the `split` string util function, that would be the best fit here for splitting a string by some fixed token (commas in your case).